indico/modules/events/tracks/controllers.py

```diff
@@ -63,3 +68,39 @@ def _process(self):
                                 tracks=[{'id': int(t.getId()), 'title': to_unicode(t.getTitle())}
                                         for t in self._conf.getTrackList()])
         return jsonify_form(form)
+
+
+class RHEditTrack(RHManageTracksBase):
+
+    def _checkParams(self, params):
+        RHConferenceModifBase._checkParams(self, params)
+        self.track = Track.find_one(id=request.view_args['id'])
+
+    def _process(self):
+        form = TrackForm(obj=FormDefaults(self.track))
+        if form.validate_on_submit():
+            update_track(self.track, form.data)
+            return jsonify_data(flash=False)
+        return jsonify_form(form)
+
+
+class RHSortTracks(RHManageTracksBase):
+
+    def _process(self):
+        sort_order = request.json['sort_order']
+        tracks = {t.id: t for t in self.event_new.tracks}
+        for position, track_id in enumerate(sort_order, 1):
+            if track_id in tracks:
+                tracks[track_id].position = position
+
+
+class RHDeleteTrack(RHManageTracksBase):
+
+    def _checkParams(self, params):
+        RHConferenceModifBase._checkParams(self, params)
+        self.track = Track.find_one(id=request.view_args['id'])
```

Comment:
You get the track in two different request handlers. What about adding a `RHManageTrackBase` that does this for you automatically?

```python
class RHManageTrackBase(RHManageTracksBase):
    def _checkParams(self, params):
        RHManageTracksBase._checkParams(self, params)
        self.track = Track.find_one(id=request.view_args['id'])


class RHEditTrack(RHManageTrackBase):
    def _process(self):
        # ...


class RHDeleteTrack(RHManageTrackBase):
    def _process(self):
        # ...
```